src/wolfsftp.c

```diff
@@ -2695,24 +2663,33 @@ static int wolfSSH_SFTPNAME_readdir(WOLFSSH* ssh, WDIR* dir, WS_SFTPNAME* out,
 static int wolfSSH_SFTPNAME_readdir(WOLFSSH* ssh, WDIR* dir, WS_SFTPNAME* out,
         char* dirName)
 {
-    int sz;
+    int sz, special = 0;
     HANDLE findHandle;
     char realFileName[MAX_PATH];
 
     if (dir == NULL || ssh == NULL || out == NULL) {
         return WS_BAD_ARGUMENT;
     }
 
-    if (*dir == INVALID_HANDLE_VALUE) {
+    /* special case of getting drives at "/" */
+    if (dirName[0] == '/' && dirName[1] == 0) {
+        if (ssh->driveIdx >= ssh->driveListCount)
+            return WS_FATAL_ERROR;
+        realFileName[0] = ssh->driveList[ssh->driveIdx++];
+        realFileName[1] = ':';
+        realFileName[2] = '\0';
+        special = 1;
+    }
+    else if (*dir == INVALID_HANDLE_VALUE) {
         char name[MAX_PATH];
         word32 nameLen = (word32)WSTRLEN(dirName);
 
-        if (nameLen > MAX_PATH - 3) {
+        if (nameLen > MAX_PATH - 2) {
             WLOG(WS_LOG_SFTP, "Path name is too long.");
             return WS_FATAL_ERROR;
         }
         WSTRNCPY(name, dirName, MAX_PATH);
-        WSTRNCAT(name, "\\*", MAX_PATH);
+        WSTRNCAT(name, "/*", MAX_PATH);
```

Comment:
It looks like this is intentional, but I just want to check that we _do_ in fact want to use / instead of \\, even though this is Windows.

Reply:
This is intentional. While back-slashes are the common directory separator in Windows, the file APIs accept paths with forward-slashes.